src/batch-submitter/batch-submitter.ts

```diff
@@ -176,12 +176,26 @@ export abstract class BatchSubmitter {
   }
 
   protected async _submitAndLogTx(
-    txFunc: (gasPrice) => Promise<TransactionReceipt>,
-    successMessage: string
+    contractFunction: Function,
+    params: any[],
+    successMessage: string,
   ): Promise<TransactionReceipt> {
     this.lastBatchSubmissionTimestamp = Date.now()
     this.log.debug('Waiting for receipt...')
 
+    const sendTransactionFunction = async (gasPrice): Promise<TransactionReceipt> => {
+      console.debug(contractFunction)
+      params.push(gasPrice)
+      const tx = await contractFunction(...params)
+      // const tx = await contractFunction(...params, {gasPrice})
```

Comment:
Is passing the `{gasPrice}` not working? That is the correct syntax for passing overwrites to a contract call

Reply:
This could explain why the tx isn't sending again, since it would resend with the same gas price and be rejected by the network. You need to increase the gas price by 15% generally to "replace by fee" in ethereum

Reply:
tried both the commented out code and as is, both don't send the tx for some reason. i've logged it all the way up to the `send` and it looks ok until then so pretty confused